### PR TITLE
Fix LIKE conditions for Array and SQL persistences

### LINKED ISSUE #530: Conditions on boolean column type work unexpectedly
When boolean field is added to a model like:
```
$m->addField('payment_is_paid', ['type' => 'boolean', 'required' => true]);
```

Then when condition like:
```
$q = '20201093';
$cond = [];
$cond[] = ['payment_is_paid', 'like', '%' . $q . '%'];
$m->addCondition($cond);
```
is added it filters all records with the flag set to 0, here is a part of the generated query:
```
... `inv`.`payment_is_paid` like 0 ...
```
instead of:
```
... `inv`.`payment_is_paid` like '%20201093%' ...
```
<br><br>Check the behaviour of integer and date(time) types too.

### PR DESCRIPTION
**Pull request by DarkSide666**
fix #530

### CODE REVIEW

File: src/Model.php [hunk 1]
```diff
@@ -1228,7 +1228,7 @@ public function addCondition($field, $operator = null, $value = null)
         $f = is_string($field) ? $this->getField($field) : ($field instanceof Field ? $field : false);
         if ($f) {
             if ($operator === '=' || func_num_args() == 2) {
-                $v = $operator === '=' ? $value : $operator;
+                $v = ($operator === '=' ? $value : $operator);
```
Comment (mvorisek) [inline src/Model.php]: brackets are not needed

File: src/Model.php [hunk 2]
```diff
@@ -1178,7 +1178,7 @@ public function removeAction($name)
      *  ->addCondition('my_field', '!=', $value);
      *  ->addCondition('my_field', 'in', [$value1, $value2]);
      *
-     * Second argument could be '=', '>', '<', '>=', '<=', '!=' or 'in'.
+     * Second argument could be '=', '>', '<', '>=', '<=', '!=', 'in' or 'like'.
```
Comment (mvorisek) [inline src/Model.php]: fix for regexp as well

File: src/Action/Iterator.php [hunk 3]
```diff
@@ -66,12 +66,12 @@ public function like($field, $value)
         $this->generator = new \CallbackFilterIterator($this->generator, function ($row) use ($field, $value) {
 
             // skip row. does not have field at all
-            if (!isset($row[$field])) {
+            if (!array_key_exists($field, $row)) {
                 return false;
             }
 
             $clean_value = trim(trim($value), '%');
```
Comment (mvorisek) [inline src/Action/Iterator.php]: The trims are not compatible with the swith below.

File: src/Persistence/SQL.php [hunk 4]
```diff
@@ -441,7 +440,8 @@ public function initQueryConditions(Model $m, Query $q): Query
                 }
                 $q->where($cond[0], $cond[1]);
             } else {
-                if ($cond[0] instanceof Field) {
+                // "like" condition do not need typecasting to field type!
+                if ($cond[0] instanceof Field && strtolower($cond[1]) != 'like') {
```
Comment (mvorisek) [inline src/Persistence/SQL.php]: Use `in_array` and fix for regexp as well.

File: src/Model.php (hunk 1 above)
Comment (DarkSide666) [reply src/Model.php]: I know. Added them for readability. I really don't like when there are to many `=` in one line :)

Comment (mvorisek) [reply src/Model.php]: Once PSR-12 CS rules will be implemented I think they will be removed :)

File: src/Action/Iterator.php (hunk 3 above)
Comment (DarkSide666) [reply src/Action/Iterator.php]: Why not? Trim gives us clean value which is used below.

File: src/Persistence/SQL.php (hunk 4 above)
Comment (DarkSide666) [reply src/Persistence/SQL.php]: ok, fixed

File: src/Model.php (hunk 2 above)
Comment (DarkSide666) [reply src/Model.php]: ok fixed

File: src/Action/Iterator.php (hunk 3 above)
Comment (mvorisek) [reply src/Action/Iterator.php]: As the `substr`s in the switch are checking in untrimmed string.

Comment (DarkSide666) [reply src/Action/Iterator.php]: Ah, right. Fixed in https://github.com/atk4/data/pull/538

**Pull request merged**